api/internal/core/store/store_mock_test.go

```diff
@@ -0,0 +1,200 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package store
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/apisix/manager-api/internal/core/entity"
+	"github.com/apisix/manager-api/internal/utils"
+	"github.com/shiningrush/droplet/data"
+	"github.com/stretchr/testify/assert"
+)
+
+type testObj struct {
+	entity.BaseInfo
+	Name  string
+	Value string
+}
+
+func keyFunc(obj interface{}) string {
+	return utils.InterfaceToString(obj.(*testObj).Name)
+}
+
+func TestMock_Create(t *testing.T) {
+	ctx := context.Background()
+	m := NewMock(keyFunc, nil)
+
+	// create success
+	ret, err := m.Create(ctx, &testObj{Name: "test"})
+	assert.Nil(t, err)
+	assert.Equal(t, "test", ret.(*testObj).Name)
+
+	// create failed, "key is required"
+	ret, err = m.Create(ctx, &testObj{Name: ""})
+	assert.Equal(t, fmt.Errorf("key is required"), err)
```

Comment:
1. Maybe `assert.Error` method is more proper.
2. Use `errors.New`.

Reply:
Updated.